engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/content-hosts-bulk-traces-modal.controller.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.content-hosts.controller:ContentHostsBulkTracesController
+ *
+ * @requires $scope
+ * @requires $state
+ * @requires $timeout
+ * @requires $window
+ * @requires $uibModalInstance
+ * @requires HostBulkAction
+ * @requires Organization
+ * @requires CurrentOrganization
+ * @requires Notification
+ * @requires Nutupane
+ * @requires BastionConfig
+ * @requires HostTraces
+ * @requires hostIds
+ *
+ * @description
+ *   Provides the functionality for the content host package list and actions.
+ */
+/*jshint camelcase:false*/
+angular.module('Bastion.content-hosts').controller('ContentHostsBulkTracesController',
+    ['$scope', '$state', '$timeout', '$window', '$uibModalInstance', 'HostBulkAction', 'Organization', 'CurrentOrganization', 'Notification', 'Nutupane', 'BastionConfig', 'HostTraces', 'hostIds',
+    function ($scope, $state, $timeout, $window, $uibModalInstance, HostBulkAction, Organization, CurrentOrganization, Notification, Nutupane, BastionConfig, HostTraces, hostIds) {
+
+        var tracesNutupane = new Nutupane(HostBulkAction, hostIds, 'traces');
+        tracesNutupane.enableSelectAllResults();
+        tracesNutupane.masterOnly = true;
+        $scope.controller = 'host_traces';
+        $scope.table = tracesNutupane.table;
+        $scope.remoteExecutionPresent = BastionConfig.remoteExecutionPresent;
+        $scope.tracesearch = false;
+
+        $scope.traceActionFormValues = {
+            authenticityToken: $window.AUTH_TOKEN.replace(/&quot;/g, '')
+        };
+
+        $scope.selectedTraceHelpers = function() {
```

Comment:
I'm curious about the behavior here:

If i have two hosts and one of them needs a reboot, will they both end up getting rebooted?

Similarly: if I have two hosts each with different traces, will only the relevant services get restarted on the respective hosts?

Reply:
@jturel good catch it looks like it does not account for that.

Here is a screenshot of me selecting 1 host with a service and 1 host with a reboot:

https://nimb.ws/e8NMA5

And here is what REX is giving us:

https://nimb.ws/bq6OHK

I am guessing that method works fine for a single host which is what I copied from, so I will have to work to adjust it. Would the errata bulk action be a good example to look at or do you suggest a different one?

Reply:
Fixed in the upcoming revision with your help